pmd-core/src/main/java/net/sourceforge/pmd/util/datasource/internal/PathDataSource.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.datasource.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+import net.sourceforge.pmd.internal.util.ShortFilenameUtil;
+
+/**
+ * DataSource implementation to read data from a {@link java.nio.file.Path}.
+ * This can also be a Path inside a zip file.
+ */
+public class PathDataSource extends AbstractDataSource {
```

Comment:
considering that `FileDataSource` is deprecated in favor of `TextFile`, shouldn't this be deprecated too? As this is a new class in PMD 7, do we even need this new class introduced at this point and just have a `TextFile`-related alternative?

Reply:
Maybe we should add a TODO to not forget to clean this up (means - delete in PMD 7).

I could imagine, that we really don't need in PMD 7 all the "DataSource" implementations anymore.